python/src/HSICStatImplementation_doc.i.in

```diff
@@ -0,0 +1,86 @@
+%define OT_HSICStat_doc
+"Base class of HSICStat.
+
+Available constructors:
+    HSICStat()
+
+Notes
+-----
+This is a base class for the implementation of an HSIC statistic
+but cannot be used by itself. An HSIC statistic object can created by using the
+derived classes: :class:`~openturns.HSICUStat` and :class:`~openturns.HSICVStat`.
+
+See also
+--------
+HSICUStat, HSICVStat"
+%enddef
+%feature("docstring") OT::HSICStatImplementation
+OT_HSICStat_doc
+
+// ---------------------------------------------------------------------
+
+%define OT_HSICStat_computeHSICIndex_doc
+"Compute the HSIC index between two samples.
+
+Parameters
+----------
+inSample : Sample
```

Comment:
```suggestion
inSample : :class:`~openturns.Sample`
```
Links to the relevant API docs must be created for parameter types. Please check all parameters.